sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/healthcare/HL7v2IO.java

```diff
@@ -596,9 +664,14 @@ public void listMessages(
           lastClaimedMilliSecond = cursor.getMillis();
         }
 
-        outputReceiver.output(msg);
+        switch (timestampMethod) {
+          case SEND_TIME:
+            outputReceiver.outputWithTimestamp(msg, Instant.parse(msg.getSendTime()));
+            break;
+          case INPUT_ELEMENT:
+            outputReceiver.outputWithTimestamp(msg, context.timestamp());
```

Comment:
You should use output(msg) and not outputWithTimestamp since output() by default does this. This also allows you to not have to take a ProcessContext parameter.